### ISSUE TITLE
[BUG] Data doesn't refresh When Nested

### ISSUE THREAD

**Issue by xrchennaidev** [mackysoft/Unity-SerializeReferenceExtensions#62]
### What happened?

So when i nest any class or interface and have this attribute ,the data inside doesnt refresh when i add element to that list ,also it somehow changes the data of the previous element also in array or list .
<img width="585" alt="BUg" src="https://github.com/user-attachments/assets/8727812e-0c93-4e02-a9c8-09b8d4fc0054">

When the above Array is serialize and you add 1 data and fill the info then click on plus to add 1 more element this new data which just got added doesnt refreshes also if you change the data in this one it also changes in first one,until you manually change type to null and back what you want.

### XPool Version

1.3.1

### Unity Version

2022.3.39f

**Comment by mackysoft** [comment]
Hi. Am I correct in assuming that you are talking about the issue of copying instances when using arrays?

In case of that problem, the following solution is recommended, or use ReorderableList and initialize the instance when adding list elements.
https://github.com/mackysoft/Unity-SerializeReferenceExtensions/issues/10

In the past, we have tried to monitor the addition of array elements in a SerializedProperty and initialize the added instance, but unfortunately, monitoring array elements in a SerializedProperty was difficult and we had to forgo it.
